<?xml version="1.0" encoding="UTF-8"?>
<!-- Generated by org.testng.reporters.JUnitReportReporter -->
<testsuite hostname="DESKTOP-9I87QRH" failures="0" tests="3" name="Nhiem.SanPham.themCTSTTest" time="0.144" errors="1" timestamp="2024-03-31T18:01:34 ICT" skipped="0">
  <testcase classname="Nhiem.SanPham.themCTSTTest" name="TC_THEMCTSP_02" time="0.047"/>
  <system-out/>
  <testcase classname="Nhiem.SanPham.themCTSTTest" name="TC_THEMCTSP_03" time="0.043"/>
  <system-out/>
  <testcase classname="Nhiem.SanPham.themCTSTTest" name="TC_THEMCTSP_01" time="0.054">
    <error message="com.microsoft.sqlserver.jdbc.SQLServerException: Violation of PRIMARY KEY constraint &#039;PK__ChiTietS__1E4FCECD6073457E&#039;. Cannot insert duplicate key in object &#039;dbo.ChiTietSanPham&#039;. The duplicate key value is (SP001-2)." type="java.lang.RuntimeException">
      <![CDATA[java.lang.RuntimeException: com.microsoft.sqlserver.jdbc.SQLServerException: Violation of PRIMARY KEY constraint 'PK__ChiTietS__1E4FCECD6073457E'. Cannot insert duplicate key in object 'dbo.ChiTietSanPham'. The duplicate key value is (SP001-2).
at util.ConnectDB.executeUpdate(ConnectDB.java:144)
at util.ChiTietSanPhamDAO.insert(ChiTietSanPhamDAO.java:24)
at view.SanPham.ThemCTSP_Test(SanPham.java:330)
at Nhiem.SanPham.themCTSTTest.TC_THEMCTSP_01(themCTSTTest.java:19)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:578)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
Caused by: com.microsoft.sqlserver.jdbc.SQLServerException: Violation of PRIMARY KEY constraint 'PK__ChiTietS__1E4FCECD6073457E'. Cannot insert duplicate key in object 'dbo.ChiTietSanPham'. The duplicate key value is (SP001-2).
at com.microsoft.sqlserver.jdbc.SQLServerException.makeFromDatabaseError(SQLServerException.java:265)
at com.microsoft.sqlserver.jdbc.SQLServerStatement.getNextResult(SQLServerStatement.java:1695)
at com.microsoft.sqlserver.jdbc.SQLServerPreparedStatement.doExecutePreparedStatement(SQLServerPreparedStatement.java:620)
at com.microsoft.sqlserver.jdbc.SQLServerPreparedStatement$PrepStmtExecCmd.doExecute(SQLServerPreparedStatement.java:539)
at com.microsoft.sqlserver.jdbc.TDSCommand.execute(IOBuffer.java:7685)
at com.microsoft.sqlserver.jdbc.SQLServerConnection.executeCommand(SQLServerConnection.java:4048)
at com.microsoft.sqlserver.jdbc.SQLServerStatement.executeCommand(SQLServerStatement.java:272)
at com.microsoft.sqlserver.jdbc.SQLServerStatement.executeStatement(SQLServerStatement.java:246)
at com.microsoft.sqlserver.jdbc.SQLServerPreparedStatement.executeUpdate(SQLServerPreparedStatement.java:484)
at util.ConnectDB.executeUpdate(ConnectDB.java:141)
... 29 more
]]>
    </error>
  </testcase> <!-- TC_THEMCTSP_01 -->
  <system-out/>
</testsuite> <!-- Nhiem.SanPham.themCTSTTest -->
